core/src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -134,6 +144,59 @@ import Prelude hiding (lines, words, unlines, unwords)
 
 #include "DocTestUnicodeStream.hs"
 
+-------------------------------------------------------------------------------
+-- Word16 to Word8 helpers
+-------------------------------------------------------------------------------
+
+data ToW16leState s x
+    = T16Nothing !s
+    | T16Just !s !x
+
+{-# INLINE scanWord16le #-}
+scanWord16le :: Monad m => Stream m Word8 -> Stream m Word16
+scanWord16le (D.Stream step state) = D.Stream step1 (T16Nothing state)
```

Comment:
We can have two separate functions, one to convert Word8 stream to Word16, another one to swap the byte order. Because a lot of times it is more efficient to generate Word16 stream directly from arrays rather than combining words.